src/crypto/PSASessionKeystore.cpp

```diff
@@ -24,62 +24,54 @@ namespace Crypto {
 
 namespace {
 
-class AesKeyAttributes
+class keyAttributesBase
 {
 public:
-    AesKeyAttributes()
+    keyAttributesBase(psa_key_id_t keyId, psa_key_type_t type, psa_algorithm_t algorithm, psa_key_usage_t usageFlags, size_t bits)
     {
-        constexpr psa_algorithm_t kAlgorithm = PSA_ALG_AEAD_WITH_AT_LEAST_THIS_LENGTH_TAG(PSA_ALG_CCM, 8);
-
-        psa_set_key_type(&mAttrs, PSA_KEY_TYPE_AES);
-        psa_set_key_algorithm(&mAttrs, kAlgorithm);
-        psa_set_key_usage_flags(&mAttrs, PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
-        psa_set_key_bits(&mAttrs, CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES * 8);
+        psa_set_key_type(&mAttrs, type);
+        psa_set_key_algorithm(&mAttrs, algorithm);
+        psa_set_key_usage_flags(&mAttrs, usageFlags);
+        psa_set_key_bits(&mAttrs, bits);
+        if (to_underlying(KeyIdBase::Maximum) >= keyId && keyId >= to_underlying(KeyIdBase::Minimum))
+        {
+            psa_set_key_lifetime(&mAttrs, PSA_KEY_LIFETIME_PERSISTENT);
+            psa_set_key_id(&mAttrs, keyId);
+        }
```

Comment:
I'm not sure this is the best approach to configure the persistance of a PSA key. This seems to be "hidding" the persistance configuration behind the KeyID.

When we transitionned the ICD monitoring entries to be able to use PSA, we had established that a new API should be added to the SymmetricKeyStore (SessionKeyStore rename not done yet). See https://github.com/project-chip/connectedhomeip/issues/30718#issuecomment-1839093000

Reply:
> I'm not sure this is the best approach to configure the persistance of a PSA key. This seems to be "hidding" the persistance configuration behind the KeyID.
> 
That is true, but all key IDs that are within `KeyIdBase::Minimum` and `KeyIdBase::Maximum` **should** be persistent. This is because that range is defined as matter-specific persistent key range. AFAIK volatile keys are created in a different range, but will check that. It is also crucial not to modify this key range as it will break backwards compatibility.

My intention of this implementation was to leave as little room for error as possible. this way we allow setting persistence only for the given range, and all keys that should be persisted are defined in one place, under `KeyIdBase` https://github.com/project-chip/connectedhomeip/pull/34925/files#diff-b7b07e88bffdc16e60f9b000afbc1eba7e2ff172b54bb9701df597ee01da77acR66 to avoid possibility of a conflict between two the same Key IDs